Improve blockchain/README.md (#9630)
TheAlgorithms/Python: blockchain/README.md

# Blockchain

A Blockchain is a type of **distributed ledger** technology (DLT) that consists of a growing list of records, called **blocks**, that are securely linked together using **cryptography**.

Let's break down the terminologies in the above definition. We find below terminologies,

- Digital Ledger Technology (DLT)
- Blocks
- Cryptography

## Digital Ledger Technology

>  It is otherwise called distributed ledger technology. It is simply the opposite of a centralized database. Firstly, what is a **ledger**? A ledger is a book or collection of accounts that records account transactions.

Reviewer: ```suggestion
Digital ledger technology is also called distributed ledger technology. It is simply the opposite of a centralized database. Firstly, what is a **ledger**? A ledger is a book or collection of accounts that records account transactions.
```

Author: Didn't like the sentence formation

>  *Why is Blockchain addressed as a digital ledger if it can record more than account transactions? What other transaction details and information can it hold?*

Reviewer: ```suggestion
*Why is Blockchain addressed as a digital ledger if it can record more than account transactions? What other transaction details and information can it hold?*
```

> Digital Ledger Technology is just a ledger that is shared among multiple nodes. This way there exists no need for central authority to hold the info. Okay, how is it differentiated from a central database and what are their benefits?

Reviewer: ```suggestion
Digital Ledger Technology is just a ledger that is shared among multiple nodes. This way there exists no need for a central authority to hold the info. Okay, how is it differentiated from a central database and what are their benefits?
```

> There is an organization that has 4 branches whose data are stored in a centralized database. So even if one branch needs any data from the ledger it needs approval from the database in charge. And if one hacks the central database he gets to tamper and control all the data.

Reviewer: ```suggestion
Suppose that there is an organization that has 4 branches whose data are stored in a centralized database. So even if one branch needs any data from the ledger it needs approval from the database in charge. And if one hacks the central database he gets to tamper and control all the data.
```

Now let's assume every branch has a copy of the ledger and then once anything is added to the ledger by any branch it is gonna automatically reflect in all other ledgers available in other branches. This is done using a peer-to-peer network.

This means that even if information is tampered with in one branch we can find out. If one branch is hacked we can be alerted, so we can safeguard other branches. Now, assume these branches as computers or nodes and the ledger is a transaction record or digital receipt. If one ledger is hacked in a node we can detect since there will be a mismatch in comparison with other node information. So this is the concept of Digital Ledger Technology.

*Is it required for all nodes to have access to all information in other nodes? Wouldn't this require enormous storage space in each node?*

## Blocks

> In short, a block is nothing but a collection of records with a labelled header. These are connected cryptographically. Once a new block is added to a chain, the previous block is connected, more precisely said as locked, and hence, will remain unaltered. We can understand this concept once we get a clear understanding of the working mechanism of blockchain.

Reviewer: ```suggestion
In short, a block is nothing but a collection of records with a labelled header. These are connected cryptographically. Once a new block is added to a chain, the previous block is connected, more precisely said as locked, and hence will remain unaltered. We can understand this concept once we get a clear understanding of the working mechanism of blockchain.
```

Author: Punctuation needed for better flow of sentences.

## Cryptography

> It is the practice and study of secure communication techniques amid adversarial behavior. More broadly, cryptography is the creation and analysis of protocols that prevent third parties or the general public from accessing private messages.

Reviewer: ```suggestion
Cryptography is the practice and study of secure communication techniques amid adversarial behavior. More broadly, cryptography is the creation and analysis of protocols that prevent third parties or the general public from accessing private messages.
```
